transform.js

```diff
@@ -39,23 +50,40 @@ function transform (filename, opts) {
 
     // transform all detected nodes and
     // close stream when done
-    mapLimit(nodes, Infinity, iterate, function () {
+    mapLimit(nodes, Infinity, iterate, function (err) {
+      if (err) return this.emit('error', err)
       self.push(ast.toString())
       self.push(null)
     })
 
     // asynchronously update css and apply to node
+    // or exorcise to file
     function iterate (args, done) {
       const transform = args[0]
       const node = args[1]
       transform(function (err, css, prefix) {
         if (err) return done(err)
-        const str = [
-          "((require('insert-css')(" + JSON.stringify(css) + ')',
-          ' || true) && ' + JSON.stringify(prefix) + ')'
-        ].join('')
-        node.update(str)
-        done()
+        if (opts.out) {
+          // exorcise to external file
+          const dirname = path.dirname(opts.out)
+          mkdirp(dirname, function (err) {
+            if (err) return done(err)
+            const ws = (typeof opts.out === 'string')
+              ? fs.createWriteStream(opts.out)
+              : opts.out
+            eos(ws, done)
+            node.update('0')
+            ws.end(css)
```

Comment:
does this work when using browserify to bundle multiple files? from my understanding, the transform will be created once for every file being transformed, which means a new writeable stream will be created for every file in the bundle. [i came up against this](https://github.com/sheetify/sheetify/issues/34#issuecomment-172820982) when experimenting with [sheetify/bundle-css](https://github.com/sheetify/bundle-cssify).
which is why [i went to the directory approach](https://github.com/sheetify/bundle-cssify/commit/22ac32c80fafdf39d73573421f2877775c224473).